tests/integration/snowflake/test_snowflake_adapter.py

```diff
@@ -0,0 +1,347 @@
+           sf_adapter.polymorphic_constraint_statement(relation,
+                                                       True,
+                                                       LOCAL_KEY,
+                                                       REMOTE_KEY,
+                                                       LOCAL_TYPE,
+                                                       TYPE_MATCH_VAL)
+
+
+def test_build_conn_string(sf_adapter):
+    conn_string = sf_adapter._build_conn_string()
+    DATABASE = sf_adapter.credentials.database
+    USER = sf_adapter.credentials.user
+    PASSWORD = sf_adapter.credentials.password
+    ACCOUNT = sf_adapter.credentials.account
+    ROLE = sf_adapter.credentials.role
+
+    assert str(conn_string) == f'snowflake://{USER}:{PASSWORD}@{ACCOUNT}/{DATABASE}/?role={ROLE}'
+
+
+def test_count_query(sf_adapter):
+    DATABASE, SCHEMA, TABLE = "SNOWSHU_DEVELOPMENT", "POLYMORPHIC_DATA", "PARENT_TABLE"
+    query = f'SELECT * FROM {DATABASE}.{SCHEMA}.{TABLE}'
+    assert sf_adapter._count_query(query) == 14
+
+
+def test_get_relations_from_database(sf_adapter):
+    SCHEMA_OBJ = BaseSourceAdapter._DatabaseObject("POLYMORPHIC_DATA",
+                                                   Relation("snowshu_development", "POLYMORPHIC_DATA", "", None, None))
+    relations_list = [Relation("snowshu_development", "POLYMORPHIC_DATA", "PARENT_TABLE_2", mz.TABLE, None),
+                      Relation("snowshu_development", "POLYMORPHIC_DATA", "CHILD_TYPE_2_ITEMS", mz.TABLE, None),
+                      Relation("snowshu_development", "POLYMORPHIC_DATA", "PARENT_TABLE", mz.TABLE, None),
+                      Relation("snowshu_development", "POLYMORPHIC_DATA", "CHILD_TYPE_1_ITEMS", mz.TABLE, None),
+                      Relation("snowshu_development", "POLYMORPHIC_DATA", "CHILD_TYPE_0_ITEMS", mz.TABLE, None)]
+    received_relations_list = sf_adapter._get_relations_from_database(schema_obj=SCHEMA_OBJ)
+    relations_list.sort(key=lambda relation_item: relation_item.name)
+    received_relations_list.sort(key=lambda relation_item: relation_item.name)
+
+    assert received_relations_list == relations_list
+
+
+def test_sample_statement_from_relation(sf_adapter):
+    DATABASE, SCHEMA, TABLE = "SNOWSHU_DEVELOPMENT", "EXTERNAL_DATA", "ADDRESS_REGION_ATTRIBUTES"
+    relation = Relation(database=DATABASE,
+                        schema=SCHEMA,
+                        name=TABLE,
+                        materialization=TABLE,
+                        attributes=[])
+    sample = sf_adapter.sample_statement_from_relation(relation, BernoulliSampleMethod(10, units="probability"))
+
+    assert query_equalize(sample) == query_equalize(f"""
+        SELECT
+            *
+        FROM 
+            {DATABASE}.{SCHEMA}.{TABLE}
+            SAMPLE BERNOULLI (10)
+        """)
+
+    assert len(sf_adapter._safe_query(sample)) > 0
+
+
+def test_analyze_wrap_statement(sf_adapter):
```

Comment:
This test should be able to be issued against snowflake since it's a fully formed query.

Reply:
Added by Halyna on line 284. Also I believe this also covers your comment "Looks good! The only outstanding thing I saw was one of the comments on test_analyze_wrap_statement, which currently isn't issued against snowflake, but should be for the integration test."?

Reply:
Yep.